metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -981,29 +1222,101 @@ def get_schema_metadata_for_custom_sql(
         )
         return schema_metadata
 
-    def _get_project(self, node):
-        if node.get("__typename") == "EmbeddedDatasource" and node.get("workbook"):
-            return node["workbook"].get("projectName")
-        elif node.get("__typename") == "PublishedDatasource":
-            return node.get("projectName")
+    def _get_published_datasource_project_luid(self, ds):
+        if (
+            ds.get(tableau_constant.LUID)
+            and ds[tableau_constant.LUID] in self.datasource_project_map.keys()
+            and self.datasource_project_map[ds[tableau_constant.LUID]]
+            in self.tableau_project_registry
+        ):
+            return self.datasource_project_map[ds[tableau_constant.LUID]]
+
+        logger.debug(
+            f"published datasource {ds.get(tableau_constant.NAME)} project_luid not found"
+        )
+
         return None
 
+    def _get_embedded_datasource_project_luid(self, ds):
+        if (
+            ds.get(tableau_constant.WORKBOOK)
+            and ds.get(tableau_constant.WORKBOOK).get(tableau_constant.PROJECT_LUID)
+            and ds[tableau_constant.WORKBOOK][tableau_constant.PROJECT_LUID]
+            in self.tableau_project_registry
+        ):
+            return ds[tableau_constant.WORKBOOK][tableau_constant.PROJECT_LUID]
+
+        logger.debug(
+            f"embedded datasource {ds.get(tableau_constant.NAME)} project_luid not found"
+        )
+
+        return None
+
+    def _get_datasource_project_luid(self, ds):
+        # Only published and embedded data-sources are supported
+        ds_type: Optional[str] = ds.get(tableau_constant.TYPE_NAME)
+        if ds_type not in (
+            tableau_constant.PUBLISHED_DATA_SOURCE,
+            tableau_constant.EMBEDDED_DATA_SOURCE,
+        ):
+            logger.debug(
+                f"datasource {ds.get(tableau_constant.NAME)} type {ds.get(tableau_constant.TYPE_NAME)} is "
+                f"unsupported"
+            )
+            return None
+
+        func_selector: Any = {
+            tableau_constant.PUBLISHED_DATA_SOURCE: self._get_published_datasource_project_luid,
+            tableau_constant.EMBEDDED_DATA_SOURCE: self._get_embedded_datasource_project_luid,
+        }
+
+        return func_selector[ds_type](ds)
+
+    @staticmethod
+    def _get_datasource_project_name(ds: dict) -> Optional[str]:
+        if ds.get(
+            tableau_constant.TYPE_NAME
+        ) == tableau_constant.EMBEDDED_DATA_SOURCE and ds.get(
+            tableau_constant.WORKBOOK
+        ):
+            return ds[tableau_constant.WORKBOOK].get(tableau_constant.PROJECT_NAME)
+        if ds.get(tableau_constant.TYPE_NAME) == tableau_constant.PUBLISHED_DATA_SOURCE:
+            return ds.get(tableau_constant.PROJECT_NAME)
+        return None
+
+    def _get_project_browse_path_name(self, ds):
+        if self.config.extract_project_hierarchy is False:
```

Comment:
fyi - I disagree on the readability of `is False`. We should change these to `if not self.config.extract...`